Tools/gsc/finalize_manifests.py

```diff
@@ -0,0 +1,155 @@
+                and is_ascii(filename)
+                and os.path.isfile(filename)
+                and filename != script_file):
+                trusted_files += f'sgx.trusted_files.file{num_trusted} = file:{filename}\n'
+                num_trusted += 1
+
+    print(f'Found {str(num_trusted)} files in \'{root_dir}\'.')
+
+    return trusted_files
+
+def generate_library_paths():
+    ld_paths = subprocess.check_output('ldconfig -v',
+           stderr=subprocess.PIPE, shell=True).decode().splitlines()
+
+    # Library paths start without whitespace. Libraries found under a path start with an
+    # indentation.
+    ld_paths = (line for line in ld_paths if not re.match(r'(^\s)', line))
+
+    return ''.join(ld_paths)
+
+def get_binary_path(executable):
+    path = subprocess.check_output(f'which {executable}',
+           stderr=subprocess.STDOUT, shell=True).decode()
+    return path.replace('\n', '')
+
+def generate_signature(manifest):
+    sign_process = subprocess.Popen([
+        '/graphene/signer/pal-sgx-sign',
+        '-libpal', '/graphene/Runtime/libpal-Linux-SGX.so',
+        '-key', '/graphene/signer/enclave-key.pem',
+        '-output', f'{manifest}.sgx',
+        '-manifest', manifest
+        ],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE)
+
+    _, err = sign_process.communicate()
+
+    if (sign_process.returncode != 0
+        or not os.path.exists(os.path.join(os.getcwd(), manifest + '.sgx'))):
+        print(err.decode())
+        print('Finalize manifests failed due to pal-sgx-sign failure.')
+        sys.exit(1)
+
+# Iterate over manifest file to find enclave size definition and return it
+def extract_enclave_size(manifest):
+    with open(manifest, 'r') as file:
+        for line in file:
+            if not line.strip().startswith('sgx.enclave_size'):
+                continue
+
+            tokens = line.split('=')
+            if len(tokens) != 2 or '#' in tokens[1]:
+                continue
+            return tokens[1].strip()
+
+    return '0M'
+
+ARGPARSER = argparse.ArgumentParser()
+ARGPARSER.add_argument('finalize_manifests.py', metavar='SCRIPT',
+    help='Script to be run.')
+ARGPARSER.add_argument('directory', metavar='DIRNAME',
+    help='Search the directory tree from this root for files and generate list of trusted files')
+ARGPARSER.add_argument('manifests', metavar='APP.manifest',
+    nargs='+',
+    help='Application-specific manifest files. The first manifest will be used for the entry point '
+        'of the docker image. If file does not exist, manifest will be generated without '
+        'application-specific values.')
+
+def main(args=None):
+    args = ARGPARSER.parse_args(args)
+
+    if not os.path.isdir(args.directory):
+        ARGPARSER.error(f'Could not find directory {args.directory}.')
+
+    trusted_files = generate_trusted_files(args.directory)
+    library_paths = generate_library_paths()
+    env_path = os.getenv('PATH')
```

Comment:
Can we also consider the LD_LIBARAY_PATH environment variable for the library paths set by the application docker image.